src/druglab/pipe/blocks/filter.py

```
@@ -0,0 +1,314 @@
+        self._seen.clear()
+ 
+    def _get_key(self, item) -> Optional[str]:
+        try:
+            from rdkit.Chem import MolToSmiles, MolToInchi
+            from rdkit.Chem.inchi import MolToInchiKey
+ 
+            if self.key == "smiles":
+                return MolToSmiles(item)
+            else:
+                return MolToInchiKey(item)
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)